### PR TITLE
[Material] Add splashColor to FAB and FAB ThemeData

### PR DESCRIPTION
**Pull request by johnsonmh**
## Description

This will allow users to specify the `splashColor` on their floatingActionButtons.

giphy

## Tests

I added the following tests:

* `splashColor` can be specified directly to the `FloatingActionButton`, or via the `FloatingActionButtonThemeData`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 1]
```diff
@@ -243,6 +245,11 @@ class FloatingActionButton extends StatelessWidget {
   /// Defaults to [ThemeData.hoverColor] for the current theme.
   final Color hoverColor;
 
+  /// The splash color for this [FloatingActionButton]'s [InkWell].
+  ///
+  /// Defaults to [ThemeData.splashColor] for the current theme.
```
Comment (rami-a) [inline packages/flutter/lib/src/material/floating_action_button.dart]: I know the other properties don't say this, but should this say that it fallsback to the FAB theme data and then the Themedata.splashcolor?

And the other properties should probably be updated to say this too

**Review by rami-a — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 2]
```diff
@@ -410,6 +417,9 @@ class FloatingActionButton extends StatelessWidget {
     final Color hoverColor = this.hoverColor
       ?? floatingActionButtonTheme.hoverColor
       ?? theme.hoverColor;
+    final Color splashColor = this.splashColor
+        ?? floatingActionButtonTheme.splashColor
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/floating_action_button.dart]: 2 space indent?

**Review by clocksmith — APPROVED** [review]

File: packages/flutter/lib/src/material/floating_action_button.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/floating_action_button.dart]: Done.

File: packages/flutter/lib/src/material/floating_action_button.dart (hunk 2 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/floating_action_button.dart]: Thanks for spotting, no idea how that happened!

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**